1018-modal-design-guidance.md

```diff
@@ -0,0 +1,29 @@
+---
+category: component design guidance
+state: approved
+created: 2024-10-30
+updated: 2024-10-30
+---
+
+# Modal Design Guidance
+
+The [modal](https://clarity.design/documentation/modal) component informs users about a task and presents information that requires immediate attention, ensuring that critical messages are clearly conveyed.
+
+## Guidance
+
+- Design teams **must** use modals when it is necessary to grab the user's attention for important messages, such as errors that prevent users from completing an action or actions that cannot be reversed.
+- Design teams **must** use modals to show additional information without losing the context of the parent page. This helps users keep track of what they were doing while also providing important details, making it easier for them to make informed decisions.
+- Design teams **must** use modals to break down a complex workflow into a series of simpler steps. This approach makes it easier for users to follow along and complete tasks without feeling overwhelmed by too much information at once.
+- Design teams **must** test the responsiveness of modals before selecting their size. Different [modal sizes](https://clarity.design/documentation/modal#sizes) are available, with medium being the default. This ensures the modal functions well across various screen sizes and devices.
+- Design teams **must not** use modals for non-essential information that is unrelated to the current workflow, as this can distract users and disrupt their focus on important tasks.
+- Design teams **must not** use modals for complex workflows that require additional information not included within the modal, as this can confuse users and hinder their ability to complete tasks effectively.
+- Design teams **must not** use modals over other modals, as stacking them can create confusion about context. Before considering stacked modals, teams **should** take the following actions:
+    - Seek alternatives to the second modal, such as using inline expansion within the first one. This approach helps maintain context and clarity for users.
+    - Seek alternatives to the use of modals in that scenario and explore options like inline task completion. This method keeps users engaged with the main content and allows them to complete tasks without unnecessary interruptions.
```

Comment:
```suggestion
- Design teams **should not** use stack modals, as doing so can create confusion about context. Before considering stacked modals, teams **should** take the following actions:
    - Seek alternatives to the second modal, such as using inline expansion within the first one. This approach helps maintain context and clarity for users.
    - Seek alternatives to the use of modals in that scenario and explore options like inline task completion. This method keeps users engaged with the main content and allows them to complete tasks without unnecessary interruptions.
```

This is confusing. "You must not do X. Before doing X, you should do the following ..."

Reply:
cc: @jbusta5